shenyu-client/shenyu-client-sofa/src/main/java/org/apache/shenyu/client/sofa/SofaServiceBeanPostProcessor.java

```diff
@@ -104,18 +110,41 @@ private void handler(final ServiceFactoryBean serviceBean) {
         if (AopUtils.isAopProxy(targetProxy)) {
             clazz = AopUtils.getTargetClass(targetProxy);
         }
+        final ShenyuSofaClient shenyuSofaClient = AnnotationUtils.findAnnotation(clazz, ShenyuSofaClient.class);
+        final String superPath = buildApiSuperPath(shenyuSofaClient);
+        if (superPath.contains("*")) {
+            Method[] declaredMethods = ReflectionUtils.getDeclaredMethods(clazz);
+            for (Method declaredMethod : declaredMethods) {
+                if (Objects.nonNull(shenyuSofaClient)) {
+                    publisher.publishEvent(buildMetaDataDTO(serviceBean, shenyuSofaClient, declaredMethod, superPath));
+                }
+            }
+            return;
+        }
         Method[] methods = ReflectionUtils.getUniqueDeclaredMethods(clazz);
         for (Method method : methods) {
-            ShenyuSofaClient shenyuSofaClient = method.getAnnotation(ShenyuSofaClient.class);
-            if (Objects.nonNull(shenyuSofaClient)) {
-                publisher.publishEvent(buildMetaDataDTO(serviceBean, shenyuSofaClient, method));
+            ShenyuSofaClient shenyuSofaClientAnnotation = method.getAnnotation(ShenyuSofaClient.class);
+            if (Objects.nonNull(shenyuSofaClientAnnotation)) {
+                publisher.publishEvent(buildMetaDataDTO(serviceBean, shenyuSofaClientAnnotation, method, superPath));
             }
         }
     }
 
-    private MetaDataRegisterDTO buildMetaDataDTO(final ServiceFactoryBean serviceBean, final ShenyuSofaClient shenyuSofaClient, final Method method) {
+    private String buildApiSuperPath(final ShenyuSofaClient shenyuSofaClient) {
+        if (Objects.nonNull(shenyuSofaClient) && !StringUtils.isBlank(shenyuSofaClient.path())) {
+            return shenyuSofaClient.path();
+        }
+        return "";
+    }
+
+    private MetaDataRegisterDTO buildMetaDataDTO(final ServiceFactoryBean serviceBean, final ShenyuSofaClient shenyuSofaClient, final Method method, final String superPath) {
         String appName = this.appName;
-        String path = contextPath + shenyuSofaClient.path();
+        String path;
+        if (superPath.contains("*")) {
```

Comment:
superPath will not contains("*") anymore, it had been solved in `handler(ServiceFactoryBean)`, so how about remove it ?